### PR TITLE
Expose cloud settings for cloud admin users via API

### PR DESCRIPTION
**Pull request by stugots**
This fulfills Pivotal Tracker story #95884794.

The intersection of swagger-ui, Django Rest Swagger, Django Rest Framework, and Django has some idiosyncracies. You'll see "tenant" listed as a serialization field in the GET and PUT endpoints, even though it's not a return value or query string parameter of either endpoint. I don't know how much effort is required to hide it in the API documentation.  I'd prefer to leave it alone for now, and hope it disappears after a future update of one of these packages.

Pylint:
* master today: 9.38 / 10 / 10
* this pull: 9.40 / 10 / 10

Changes:
* Expose the OS_* credentials on a GET to /user, but only for tenant admins.
* Allow the updating of OS_* credentials on a PUT to /user, but only for tenant admins.
* Unit tests for the above, plus testing for permission checks.
* In installer_fabfile, fix looping bug in the auth URL prompt.

### CODE REVIEW

File: goldstone/user/tests.py [hunk 1]
```diff
@@ -240,5 +242,309 @@ def test_change_all_fields(self):
...
+        EXPECTED_CONTENT = {"username": TEST_USER[0],
+                            "first_name": '',
+                            "last_name": '',
+                            "email": TEST_USER[1],
+                            "tenant_admin": False,
+                            "default_tenant_admin": False}
+
+        cloud_fields = {"tenant_name": "cloud name 0",
+                        "username": "abracadabra",
+                        "password": "boomlackalackalacka",
```
Comment (jxstanford) [inline goldstone/user/tests.py]: I believe this should be boomshakalacka... :-)

File: installer_fabfile.py [hunk 2]
```diff
@@ -293,16 +298,21 @@ def cloud_init(gs_tenant,
...
+
+                if re.search(AUTH_URL_VERSION_LIKELY, stack_auth_url[-9:]):
+                    # The user shouldn't have included the version segment, but
+                    # did anyway. Remove it.
+                    version_index = re.search(AUTH_URL_VERSION_LIKELY,
+                                              stack_auth_url)
+                    stack_auth_url = stack_auth_url[:version_index.start()]
+
+                # Append our version number to the base URL.
+                stack_auth_url = os.path.join(stack_auth_url, AUTH_URL_VERSION)
```
Comment (jxstanford) [inline installer_fabfile.py]: The intent of the changed code block was to allow the user to copy/paste the entire value of OS_AUTH_URL from their cloud generated keystonerc_admin file.  I believe that approach will be less error prone than requiring them to input the partial (base) part of the string.

Suggestion is to revert this change, but change the prompt string to "Enter OpenStack auth URL (eg: http://10.10.10.10:5000/v2.0):"

**Comment by jxstanford** [comment]
after addressing the auth_url comment, go ahead and merge.

File: installer_fabfile.py (hunk 2 above)
Comment (stugots) [inline installer_fabfile.py]: The code in master breaks when the user doesn't enter a version number. Or, enters any version number other than exactly "2.0" or "3":

screen shot 2015-06-04 at 7 41 25 am

screen shot 2015-06-04 at 7 43 18 am


The code in this pull request will work regardless of whether the entered URL has a version number segment, and regardless of what that version number is.

Comment (jxstanford) [inline installer_fabfile.py]: Ah, got it.  please remove the term 'base' from the prompt so we are just asking  the user for the auth URL.

Comment (stugots) [inline installer_fabfile.py]: That's already been done, on line 304. Is there another spot in the code that I missed?

**Comment by lexjacobs** [comment]
:+1:

**Pull request merged**